itest/tests/zar.test.js

```diff
@@ -0,0 +1,85 @@
+/* @flow */
+
+// This is needed to use zealot outside a browser.
+global.fetch = require("node-fetch")
+
+import {exec, execSync} from "child_process"
+import path from "path"
+
+import zealot from "../../src/js/services/zealot"
+
+import {nodeZqDistDir} from "../lib/env"
+import {handleError, stdTest} from "../lib/jest.js"
+import {
+  ingestFile,
+  newAppInstance,
+  startApp,
+  waitForNewTab,
+  waitForResults
+} from "../lib/app"
+
+describe("Zar tests", () => {
+  let app
+  let testIdx = 0
+  const ZAR = path.join(nodeZqDistDir(), "zar")
+  beforeEach(() => {
+    app = newAppInstance(path.basename(__filename), ++testIdx)
+    return startApp(app)
+  })
+
+  afterEach(async () => {
+    if (app && app.isRunning()) {
+      return await app.stop()
+    }
+  })
+
+  stdTest(`Brim starts when a Zar space is present`, (done) => {
+    ingestFile(app, "sample.tsv")
+      .then(async () => {
+        await waitForResults(app)
+
+        // Figure out underlying space directories. One will be a
+        // zarRoot. The other will be used to read zng into zar.
+        const client = zealot.client("localhost:9867")
+
+        const sampleSpace = (await client.spaces.list())[0]
+        const zarSpace = await client.spaces.create({name: "sample.zar"})
+
+        const zngFile = path.join(sampleSpace.data_path, "all.zng")
+        const zarRoot = zarSpace.data_path
+
+        // Now that the spaces are read, stop the app so we can convert
+        // zarSpace to an archive store and let the app read it when it
+        // restarts.
+        await app.stop()
+
+        // Create a zar archive inside the space and index some
+        // interesting stuff.
+        execSync(`"${ZAR}" import -s 1024B -R "${zarRoot}" "${zngFile}"`)
+        await Promise.all(
+          [":ip", ":port", "uid", "_path"].map((index) =>
+            exec(`"${ZAR}" index -R "${zarRoot}" ${index}`)
+          )
+        )
+
+        // Restart the app so that it reads the new space.
+        await startApp(app)
+        // There's inconsistency here on what page is presented when the
+        // app restarts. Sometimes, it's the results viewer for
+        // sample.tsv.brim. Other times, it's the new tab page. Instead
+        // of forking off the flow further, I'm stopping any testing at
+        // this point.
+        // TODO: Understand why different pages appear when the app is
+        // restarted.
+        try {
+          await waitForResults(app)
+        } catch {
+          await waitForNewTab(app)
+        }
```

Comment:
Try a browser refresh instead.

Reply:
Considered and rejected due to other flakiness: in some instances, after issuing a https://www.electronjs.org/docs/api/browser-window#winreload, the Brim app would report the initial ingesting space as deleted. See for example here: https://github.com/brimsec/brim/runs/740557239?check_suite_focus=true

Logs show the Brim app sending the DELETE to zqd for reasons to me that are unclear.
```
[2020-06-05 00:18:56.130] [info] [6663dce9b9]:  refreshing window
[2020-06-05 00:18:56.132] [info] [6663dce9b9]:  starting delete for sp_1csTWovm9caRhn3KdnVzF2o4nDV
[2020-06-05 00:18:56.200] [info] [6663dce9b9]:  Deleted sp_1csTWovm9caRhn3KdnVzF2o4nDV
```
```
{"level":"info","ts":1591316336.133745,"logger":"http.access","msg":"Request completed","request_id":"10","host":"localhost:9867","method":"DELETE","proto":"HTTP/1.1","remote_addr":"127.0.0.1:33708","request_content_length":0,"url":"/space/sp_1csTWovm9caRhn3KdnVzF2o4nDV","elapsed":0.0001414,"response_content_length":0,"status_code":204}
```
Note that nothing in the test at  7bd212b was doing any deleting.

I don't think it's worth it to spend much more time honing this test when the zqd API is going to change again later.

Reply:
Hey Mike, when you reload the page, brim will delete all spaces that are currently ingesting. It is almost the same logic as closing the window while ingest is happening. Since reloading the page is possible but not common for users to perform, we opted for not presenting a confirmed dialogue when reloading.

Reply:
> Hey Mike, when you reload the page, brim will delete all spaces that are currently ingesting.

Thanks for pointing this out. `ingestFile()` is supposed not to return until ingest has completed, and for that, we rely on the app. Maybe something is happening in a CI environment that isn't observable locally such that that logic isn't working.

Reply:
@jameskerr I fixed this problem by waiting 10 seconds for the ingest progress indicator to appear. In some cases, the test was racing ahead of the app and moving on because the ingest progress indicator hadn't yet appeared, so the test assumed ingest was finished. 911d308 does this. I was able to reproduce the problem on a linux vm within a few iterations. After 911d308, the test past more than 2000 iterations.

Reply:
2000 iterations! Thorough